python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -259,21 +332,331 @@ def scaleModel(self, model, star, inPlace=True, nb90Rots=0):
+            )
+        return bbox, invImage
+
+    def addScaledModel(self, subtractor, brightStarStamp, multipleAnnuli=False):
+        """Add the scaled model of the given star to the subtractor plane.
+
+        Parameters
+        ----------
+        subtractor : `~lsst.afw.image.MaskedImageF`
+            The Exposure containing the scaled model of brigth stars to be subtracted from the input
+            exposure.
+        brightStarStamp : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamp`
+            The stamp of the star of which the PSF model will be scaled and added to the subtractor.
+        multipleAnnuli : bool, optional
+            If true, the model should be scaled based on a flux at a radius other than its normalization
+            radius.
+
+        Returns
+        -------
+        subtractor : `~lsst.afw.image.MaskedImageF`
+            The input subtractor Exposure with the added scaled model at the given star's location in the
+            exposure.
+        invImage: `~lsst.afw.image.MaskedImageF`
+            The extended PSF model, shifted (and potentially warped) to match the bright star's positioning.
+        """
+        bbox, invImage = self.preparePlaneModelStamp(brightStarStamp)
+        if multipleAnnuli:
+            cond = self.psf_annular_fluxes[:, 0] == brightStarStamp.optimalOuterRadius
+            psf_annular_flux = self.psf_annular_fluxes[cond, 1][0]
+            self.scaleModel(invImage,
+                            brightStarStamp,
+                            inPlace=True,
+                            nb90Rots=self.inv90Rots,
+                            psf_annular_flux=psf_annular_flux)
+        else:
+            self.scaleModel(invImage, brightStarStamp, inPlace=True, nb90Rots=self.inv90Rots)
+        # Replace NaNs before subtraction (note all NaN pixels have
+        # the NO_DATA flag).
+        invImage.image.array[np.isnan(invImage.image.array)] = 0
+        bbox.clip(self.inputExpBBox)
+        if bbox.getArea() > 0:
+            subtractor[bbox] += invImage[bbox]
+        return subtractor, invImage
+
+    def buildSubtractor(self, brightStarStamps, subtractor, invImages, multipleAnnuli=False):
+        """Build an image containing potentially multiple scaled PSF models, each at the location of a given
+        brigth star.
+
+        Parameters
+        ----------
+        brightStarStamps : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamps`
+            Set of stamps centered on each bright star to be subtracted, produced by running
+            `~lsst.pipe.tasks.processBrightStars.ProcessBrightStarsTask`.
+        subtractor : `~lsst.afw.image.MaskedImageF`
+            The Exposure that will contain the scaled model of brigth stars to be subtracted from the
+            exposure.
+        invImages : `list`
+            A list containing extended PSF models, shifted (and potentially warped) to match the bright stars
```

Comment:
Since detectors have different orientations, the PSF model should be rotated to match the orientation of the detectors in some cases. To do that, the code uses the inverse of the transform that is applied to the bright star stamp to match the orientation of the detector. I think that is why this is called `invImages`.